lib/tasks/update_cog_uk_ids_in_lighthouse_sample.rake

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+namespace :lighthouse_sample_table do
+  desc 'Update lighthouse_sample with cog uk ids from sample'
+
+  # For each positive Lighthouse Sample row, set the Cog UK ID from the matching Sample
+  task update_lighthouse_sample_cog_uk_ids: :environment do
+    rows_found = 0
+    rows_already_set = 0
+    rows_no_sample_found = 0
+    rows_sample_had_no_supplier_name = 0
+    rows_updated = 0
+
+    $stdout.puts('Updating lighthouse sample table with cog uk ids')
+
+    # Select all lighthouse samples
+    LighthouseSample.all.each do |lh_sample|
+      $stdout.puts 'Starting looping through the lighthouse samples' if rows_found.zero?
+      rows_found += 1
+      $stdout.puts "On row #{rows_found}" if (rows_found % 100).zero?
+
+      # skip if cog_uk_id already set
+      if lh_sample.cog_uk_id.present?
+        rows_already_set += 1
+        next
+      end
+
+      joins_sql = 'INNER JOIN lighthouse_sample ON sample.uuid_sample_lims = lighthouse_sample.lh_sample_uuid '\
+      "AND lighthouse_sample.lh_sample_uuid = '#{lh_sample.lh_sample_uuid}'"
+
+      samp = Sample.joins(joins_sql).where.not(
+        sample: {
+          supplier_name: nil
+        }
+      ).first
```

Comment:
So this will do 34 million queries - I guess that will be very slow?

I would lean towards doing it all in a single SQL update statement (maybe not in a rails script - not sure if there's an advantage) - what do you think?